gravitee-am-test/specs/gateway/mfa/mfa-enrollment-conditional.spec.ts

```diff
@@ -0,0 +1,294 @@
+    const loginPage = await followUpGet(authResponse, 200);
+
+    let xsrf = extractDomValue(loginPage, '[name=X-XSRF-TOKEN]');
+    let action = extractDomAttr(loginPage, 'form', 'action');
+
+    const loginPostResponse = await performFormPost(
+        action,
+        '',
+        {
+            'X-XSRF-TOKEN': xsrf,
+            username: testContext.domain.user.username,
+            password: testContext.domain.user.password,
+            rememberMe: 'off',
+            client_id: testContext.application.clientId,
+        },
+        {
+            Cookie: loginPage.headers['set-cookie'],
+            'Content-type': 'application/x-www-form-urlencoded',
+        },
+    ).expect(302);
+
+    const enrollLocationResponse = await followUpGet(loginPostResponse, 302);
+    const enrollmentPage = await followUpGet(enrollLocationResponse, 200);
+
+    xsrf = extractDomValue(enrollmentPage, '[name=X-XSRF-TOKEN]');
+    action = extractDomAttr(enrollmentPage, 'form', 'action');
+    let factorId = extractDomValue(enrollmentPage, '[name=factorId]');
+
+    const enrollmentPostResponse = await performFormPost(
+        action,
+        '',
+        {
+            'X-XSRF-TOKEN': xsrf,
+            factorId: factorId,
+            ser_mfa_enrollment: true,
+        },
+        {
+            Cookie: enrollmentPage.headers['set-cookie'],
+            'Content-type': 'application/x-www-form-urlencoded',
+        },
+    ).expect(302);
+
+    return {
+        cookie: enrollmentPostResponse.headers['set-cookie'],
+        location: enrollmentPostResponse.headers['location'],
+        factorId: factorId,
+    };
+}
+
+async function processMfaEndToEnd(rememberDevice: boolean = false) {
+    const enrollmentPostResponse = await processMfaEnrollment();
+
+    const challengeLocationResponse = await get(enrollmentPostResponse.location, 302, {Cookie: enrollmentPostResponse.cookie});
+    const challengePage = await followUpGet(challengeLocationResponse, 200);
+
+    let xsrf = extractDomValue(challengePage, '[name=X-XSRF-TOKEN]');
+    let action = extractDomAttr(challengePage, 'form', 'action');
+
+    const challengePostResponse = await performFormPost(
+        action,
+        '',
+        {
+            'X-XSRF-TOKEN': xsrf,
+            factorId: enrollmentPostResponse.factorId,
+            code: '1234',
+            rememberDeviceConsent: rememberDevice ? 'on' : 'off',
+        },
+        {
+            Cookie: challengePage.headers['set-cookie'],
+            'Content-type': 'application/x-www-form-urlencoded',
+        },
+    ).expect(302);
+
+    const finalLocationResponse = await followUpGet(challengePostResponse, 302);
+    expect(finalLocationResponse.headers['location']).toBeDefined();
+    expect(finalLocationResponse.headers['location']).toContain('code=');
+
+    return {
+        cookie: finalLocationResponse.headers['set-cookie'],
+    };
+}
```

Comment:
**quetion**: looks like this two methods are not used in this simulation. I guess we should use them at least once to be sure that a complete MFA flow is manage as expected with the defined application settings

Reply:
All specs cover e2e scenarios